### PR TITLE
Close tab keybind

### PR DESCRIPTION
**Pull request by deadcombo**
Hello,

I've added a key to close tabs. It is useful together with recency sort to clean up excursions quickly.
I've chosen w for parity with Ctrl+w.

Cheers.

### CODE REVIEW

**Comment by Rayquaza01** [comment]
Thanks for this PR!

Right now, there's no visual feedback in the popup when you close a tab. The simplest way would probably be to just remove the active element from the DOM and TabList, and set the next one down to be active.  Another way would be to mark the tab as dead (though you'd need to look up the new ID once it's closed, as open tabs don't have unique IDs from recently closed tabs)

If you'd like to add that on this PR, then go ahead! Otherwise, I can do it :)

Also, make sure it builds correctly! In the keypress check, you accidentally put a `[` instead of `{`. The command for building is `npm run build:development` or `npm run build:production`. I was going to point you to [the build instructions](https://github.com/Rayquaza01/quick-tab-switch/blob/master/BUILD.md), but I just realized that I listed the wrong command in there haha. I'll fix that tomorrow.

File: src/popup/index.ts [hunk 1]
```diff
@@ -104,6 +104,9 @@ function switchTab(e: KeyboardEvent) {
             }
             search.focus();
             active.setActive = false;
+        } else if (e.key === "w") [
```
Comment (Rayquaza01) [inline src/popup/index.ts:107]:
```suggestion
        } else if (e.key === "w") {
```

**Comment by deadcombo** [comment]
> If you'd like to add that on this PR, then go ahead! Otherwise, I can do it :)

Please do, I've tried it reading your code and MDN but I didn't figure out how to do so. I'll try to do it for my own edification with the idea you put forth but I'd rather that not hold this back.

> Also, make sure it builds correctly! In the keypress check, you accidentally put a [ instead of {. The command for building is npm run build:development or npm run build:production. I was going to point you to [the build instructions](https://github.com/Rayquaza01/quick-tab-switch/blob/master/BUILD.md), but I just realized that I listed the wrong command in there haha. I'll fix that tomorrow.

Your build instructions served well, I've been using this close tab button for a few weeks. Unfortunately I amateurishly fumbled interfacing with GitHub from my git setup and I ended up making the changes on the web editor out of late night frustration.
I hoped to be a bit more proactive but I didn't spare you any work, sorry about that 😄

**Comment by Rayquaza01** [comment]
> Something that could be improved still is the behavior when closing the tab you're in. I think it would be preferable to keep the pop-up in focus, at the moment focus is lost as Firefox switches to the previous tab, although the popup remains open. Alternatively the popup can simply close.

There is no way for an out of focus window to regain focus, without the user specifically giving it (i.e. clicking or tabbing in). Closing or changing the selected tab will cause the popup to lose focus, and AFAIK there's nothing that can be done to prevent that. I think closing the popup after closing the selected tab is probably the best option.

> Your build instructions served well, I've been using this close tab button for a few weeks. Unfortunately I amateurishly fumbled interfacing with GitHub from my git setup and I ended up making the changes on the web editor out of late night frustration. I haven't used git forges much. I hoped to be a bit more proactive but I didn't spare you any work, sorry about that 😄

Ah, it's alright. It was a pretty easy error to find and fix. It just made me a little worried that you didn't figure out how to build it haha. Glad to see I was wrong about that :)

> Please do, I've tried it reading your code and MDN but I didn't figure out how to do so. I'll try to do it for my own edification with the idea you put forth but I'd rather that not hold this back.

Alright! I'll keep this PR open for now and add the relevant changes here.

**Comment by deadcombo** [comment]
Sounds good, I've been using it since  your commits without any bugs.

I appreciate it, thank you.

**Comment by Rayquaza01** [comment]
@deadcombo Version 1.3.0 with the close tab keybind is now live on AMO.

**Comment by deadcombo** [comment]
> @deadcombo Version 1.3.0 with the close tab keybind is now live on AMO.

I noticed, thank you! Happy holidays and thanks for making my browser better!

**Pull request merged**